src/match/controllers/update_match_controller.py

```diff
@@ -0,0 +1,131 @@
+import datetime
+import json
+
+from api.utils import failure_response
+from api.utils import success_response
+from location.models import Location
+from match import match_status
+from match.models import Match
+from rest_framework import status
+
+
+class UpdateMatchController:
+    def __init__(self, match_id, user, data, serializer):
+        self._match_id = match_id
+        self._user = user
+        self._data = data
+        self._serializer = serializer
+
+    def process(self):
+        # First, find the match and any fields to be updated
+        self._match = Match.objects.filter(id=self._match_id)
+        if not self._match:
+            return failure_response("Match does not exist.")
+        self._match = self._match[0]
+        proposed_meeting_times = self._data.get("proposed_meeting_times")
+        proposed_locations = self._data.get("proposed_locations")
+        meeting_location = self._data.get("meeting_location")
+        meeting_time = self._data.get("meeting_time")
+
+        # Next, modify the attributes that may have changed
+        error_responses = []
```

Comment:
wait why are you doing this to error responses

you should just return a failure response at the very first time you encounter one

Reply:
lmk if you would rather this all be in one method or if u think there's a better way

Reply:
oh hm i see... i feel like there's a better way but i won't block you. you can always refactor after you merge 👍

Reply:
i'd review your logic with this

if frontend gets a failure they're expecting uh oh I have an error so I need to fix it before the match is updated properly but you still do updates before sending the failure response so we can easily get into an inconsistent state (i.e. regardless of it being possible, consider the case where everything is updated except my match status because I got an error and it was sent to frontend. now my proposed locations and meeting locations will be inconsistent with the current match status which was not previously updated and more errors will occur from there, making it hard to debug)

instead of checking for error responses when updating, can you just check that the data passed in by frontend is 100% correct in a helper before proceeding with updates?

Reply:
good points, i made sure to return the failure before I ever called `match.save()` so the match won't be updated but you're right that I should catch the errors earlier